### PR TITLE
logZ(beta) functionality and logX function

### LINKED ISSUE #202: logX functionality
Resurrecting #81, it would be good to be able to compute `samples.logX`. When I have thought about this in the past I remember there being breaking changes required, so it is worth considering this further before 2.0.0.

### PR DESCRIPTION
**Pull request by williamjameshandley**
# Description

This PR 

- Introduces a `NestedSamples.logX` function, which computes or simulates prior log-volumes, fixing #202
- Introduces a logw function, which is the nested sampling weights (and now does most of the heavy lifting, where previously this was done by dlogX)
- Upgrades logZ, D, d so that they are capable of taking beta as a scalar or a vector. This allows partition function outputs (fixing #3). If both beta and nsamples are provided, this is handled with some clever pandas multiindexing.
- Adds logL to the ns_outputs array (fixing #196)
- changes logs to lns in tex files (but not in code, fixing #197)

There are a couple of bugs which will be fixed by #198 when printing some of the data frames, which struggle to handle weights and multiindices.

Fixes #202 #3 #197 #196

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: anesthetic/samples.py [hunk 1]
```diff
@@ -553,71 +553,219 @@ def ns_output(self, nsamples=200):
...
+        samples['d'] = np.exp(logsumexp(logw-samples.logZ, b=(S-samples.D)**2,
+                              axis=0))*2
+        samples['logL'] = samples['logZ'] + samples['D']
 
-        samples.tex = {'logZ': r'$\log\mathcal{Z}$',
+        samples.tex = {'logZ': r'$\ln\mathcal{Z}$',
                        'D': r'$\mathcal{D}$',
-                       'd': r'$d$'}
+                       'd': r'$d$',
+                       'logL': r'$\ln L$'}
```
[the remark this answers is not in the document]
Comment (williamjameshandley) [reply anesthetic/samples.py]: Done!

File: anesthetic/samples.py [hunk 2]
```diff
@@ -553,71 +553,219 @@ def ns_output(self, nsamples=200):
...
-        S = (dlogX*0).add(self.beta * self.logL, axis=0) - samples.logZ
+        betalogL = self._betalogL(beta)
+        S = (logw*0).add(betalogL, axis=0) - samples.logZ
 
-        samples['D'] = np.exp(logsumexp(logw, b=S, axis=0))
-        samples['d'] = np.exp(logsumexp(logw, b=(S-samples.D)**2, axis=0))*2
+        samples['D'] = np.exp(logsumexp(logw-samples.logZ, b=S, axis=0))
+        samples['d'] = np.exp(logsumexp(logw-samples.logZ, b=(S-samples.D)**2,
+                              axis=0))*2
+        samples['logL'] = samples['logZ'] + samples['D']
```
[the remark this answers is not in the document]
Comment (williamjameshandley) [reply anesthetic/samples.py]: Agreed.

File: anesthetic/samples.py [hunk 3]
```diff
@@ -553,71 +553,219 @@ def ns_output(self, nsamples=200):
...
-                       'd': r'$d$'}
+                       'd': r'$d$',
+                       'logL': r'$\ln L$'}
         samples.label = self.label
         return samples
 
-    def logZ(self, nsamples=None):
-        """Log-Evidence.
+    def logX(self, nsamples=None):
+        """Log-Volume.
```
[the remark this answers is not in the document]
Comment (williamjameshandley) [reply anesthetic/samples.py:622]: Done

**Comment by williamjameshandley** [comment]
One test fails, but this will be fixed when #198 is merged.

File: anesthetic/samples.py [hunk 4]
```diff
@@ -542,82 +542,302 @@ def prior(self, inplace=False):
...
+        samples['d_G'] = s * np.exp(logd_G_o2*2)
+        samples['logL_P'] = samples['logZ'] + samples['D_KL']
 
-        samples.tex = {'logZ': r'$\log\mathcal{Z}$',
-                       'D': r'$\mathcal{D}$',
-                       'd': r'$d$'}
+        samples.tex = {'logZ': r'$\ln\mathcal{Z}$',
+                       'D_KL': r'$\mathcal{D}_\mathrm{KL}$',
+                       'd_G': r'$d_\mathrm{G}$',
+                       'logL_P': r'$\langle\ln L\rangle_\mathcal{P}$'}
```
Comment (lukashergt) [inline anesthetic/samples.py:617]: `\ln L` --> `\ln\mathcal{L}`

Comment (williamjameshandley) [reply anesthetic/samples.py]: Done

File: anesthetic/samples.py [hunk 5]
```diff
@@ -553,71 +553,219 @@ def ns_output(self, nsamples=200):
...
-        logw -= samples.logZ
-        S = (dlogX*0).add(self.beta * self.logL, axis=0) - samples.logZ
+        betalogL = self._betalogL(beta)
+        S = (logw*0).add(betalogL, axis=0) - samples.logZ
 
-        samples['D'] = np.exp(logsumexp(logw, b=S, axis=0))
-        samples['d'] = np.exp(logsumexp(logw, b=(S-samples.D)**2, axis=0))*2
+        samples['D'] = np.exp(logsumexp(logw-samples.logZ, b=S, axis=0))
+        samples['d'] = np.exp(logsumexp(logw-samples.logZ, b=(S-samples.D)**2,
+                              axis=0))*2
```
[the remark this answers is not in the document]
Comment (williamjameshandley) [reply anesthetic/samples.py]: It's about a factor of 2 faster in the new format, and it is also more readable.

**Review by lukashergt — APPROVED** [review]
Very nice.

**Review by lukashergt — APPROVED** [review]

**Pull request merged**